pkg/api/aim/runs.go

```diff
@@ -202,31 +210,43 @@ func GetRunMetrics(c *fiber.Ctx) error {
 	}
 
 	metrics := make(map[string]struct {
-		values []*float64
-		iters  []int
+		name    string
+		iters   []int
+		values  []*float64
+		context datatypes.JSON
 	}, len(metricKeys))
 	for _, m := range r.Metrics {
-		k := metrics[m.Key]
-
 		v := m.Value
 		pv := &v
 		if m.IsNan {
 			pv = nil
 		}
 
-		k.values = append(k.values, pv)
+		key := m.Key
+		if m.ContextID != nil {
+			key = fmt.Sprintf("%s%d", m.Key, *m.ContextID)
+		}
+		k := metrics[key]
```

Comment:
what is `k`?

Reply:
this moment was here before. this is how empty k(struct) initialised.